docs/proposals/loadbalancing.md

```diff
@@ -0,0 +1,311 @@
+
+The proposed long term solution for this problem is to reference a claim on an ip through the Ingress Path.
+
+### How does TLS work?
+
+Each IngressPath specifies a single secret and a tls mode. Its upto the loadbalancer controller to handle the tls mode. For example, the gce controller will only support TLS termination for 1.1 because each IngressPath gets a new loadbalancer, and GCE doesn't allow multiple certs per IP. With the current design a user can create the wrong type of secret for the wrong mode. They wouldn't know till they try creating the Ingress path. Validating a simple key:value secret is also hard.
+
+Some alternatives to the proposed tls mode handling:
+* Remove tlsMode and create a new type of secret(s).
+* Don't use a secret, embed the TLSConfig in the IngressPath.
+
+### What triggers the creation of a new loadbalancer controller, should this be exposed to users?
+
+In the current proposal, the kube-controller-manager is started up with a --loadbalancer-controllers flag that dictates which controllers to startup. This is obviously sub-optimal. A better solution would be --loadbalancer-controllers=gce-lbc-pod,haproxy-lbc-pod..., then this just boils down to creating some rcs. Taking this a step further, we could have a resource to allow dynamically adding more loadbalancer controllers to the cluster.
+
+### How does public DNS work?
+
+Most of this is TDB. With the current model, if someone specifies a hostname the loadbalancer controllers assume they know what they're doing. If there are 2 IngressPaths with the same hostname and url endpoints, one of them will win. This is just like the overlapping labels rc problem.
+
+### What does the apiserver validate vs the loadbalancer controller?
+
+Not all loadbalancer backends will support all IngressPath configuration. This will be some confusion around validation errors. The apiserver will validate api constituents and nothing more (i.e that an IngressPath points to a valid service, that the tlsmodes match up to the provided secrets etc).
+
+### How do we expose more complex loadbalancing concepts through the api?
+
+This is the most interesting question. We need initial discussion to reach a conclusion first. In this context "more advanced" is really not *that* advanced:
+* Tls modes other than termination
+* Persistent sessions
+* Loadbalancing algorithms (cloud provider support at this level is limited)
+* Loadbalancer health checks: are they even necessary given that pods have probes?
```

Comment:
Getting the OpenShift F5 router (which is currently a pending pr iirc) checked in as a loadbalancer controller to kube would be nice. Mostly because there's no way I'm going to get my hands on an F5 to write it on my own.